### ISSUE TITLE
Add support for Cilium CNI

### ISSUE THREAD

**Issue by yboaron** [submariner-io/submariner#3168]
**What would you like to be added**:

I am creating this issue to track Submariner support for Cilium CNI.

Currently, Cilium CNI is recognized as a generic CNI by Submariner, and subctl verify e2e tests [seem to fail](https://github.com/submariner-io/submariner/issues/3159#issuecomment-2352351748) on clusters with Cilium CNI.

**Why is this needed**:

Enable multi-cluster networking using Submariner on clusters with Cilium CNI.

**Other Info**:

In order to support new CNI in Submariner there is a need to update the following functions:

- [CNI network discovery](https://github.com/submariner-io/submariner-operator/blob/devel/pkg/discovery/network/network.go#L71) - responsible for pod,service CIDRs and CNI plugin name auto-discovery.
- Submariner RouteAgent handlers (optional) - used to apply configuration specific for CNI, check [Calico ippool handler](https://github.com/submariner-io/submariner/blob/devel/pkg/routeagent_driver/handlers/calico/ippool_handler.go) for example.

**Comment by dfarrell07** [comment]
@gk-fschubert agreed to help with this one in https://github.com/submariner-io/submariner/issues/3159#issuecomment-2371940830. Thanks! You can assign yourself here (I'd do it, but can't due to GitHub anti-spam restrictions).

**Comment by yboaron** [comment]
This [post](https://medium.com/@yboaron/d56d7c39f0cb) covers Submariner deployment on K8S-Kind/Cilium and K8S-Kind/Calico clusters

**Comment by dfarrell07** [comment]
We're looking for contributors to work on Cilium tasks for SubM if anyone is willing to step up.

**Comment by jdaln** [comment]
Is there anything new on this front or is it mostly waiting for someone to implement nftables to eBPF?
\+ of course what what @yboaron points out

**Comment by yboaron** [comment]
>Is there anything new on this front

Not much; we're still looking for a contributor to lead Cilium implementation for Submariner,  we’d love to see some community contributions on this task.

**Comment by jdaln** [comment]
If plans don't change, I should be looking into it in September. Happy to have people reach out if they have the same need.
